truffle/src/main/java/org/jruby/truffle/nodes/core/BigDecimalNodes.java

```diff
@@ -110,21 +168,23 @@ public RubyBasicObject initialize(RubyBasicObject self, RubyBasicObject v) {
 
         @Specialization(guards = "isRubyString(v)")
```

Comment:
I would put `@TruffleBoundary` on this. The `v.toString()` is not going to compile well.